### PR TITLE
CIL API update

### PR DESCRIPTION
**Pull request by ParnianAlimi**
Address CRYP-170. Refactor's Polymesh's use of the CIL library. Essentially removes objects to Scalar conversions.

`mercat` is the only branch that points to the cryptography's develop AFAIK, if there are others, they should cherry-pick this change as well.

The PR for changes to the CIL if you are curious: https://github.com/PolymathNetwork/cryptography/pull/85

### CODE REVIEW

File: pallets/confidential/Cargo.toml
Comment (ParnianAlimi) [inline pallets/confidential/Cargo.toml:33]: I will be reverting back to `develop` as soon as my other PR goes in.

File: polymesh_schema.json
Comment (maxsam4) [inline polymesh_schema.json:4]: Hmm, Is 128 bits of entropy enough for our use-case?

File: pallets/confidential/Cargo.toml
Comment (maxsam4) [reply pallets/confidential/Cargo.toml:33]: LGTM but I'd suggest merging this after the PR in crypto repo is merged.

**Review by maxsam4 — APPROVED** [review]

File: primitives/src/investor_zkproof_data.rs
Comment (vkomenda) [inline primitives/src/investor_zkproof_data.rs:38]: This could be camel-cased, `Cdd`.

File: polymesh_schema.json
Comment (ParnianAlimi) [reply polymesh_schema.json:4]: This is the output of PUIS. I'm tagging @JesseALancaster to comment.

File: pallets/confidential/Cargo.toml
Comment (ParnianAlimi) [reply pallets/confidential/Cargo.toml:33]: I'll make sure that happens.

**Review by satyamakgec — APPROVED** [review]

**Review by aafshar-poly — APPROVED** [review]
Nice job :+1:

**Pull request merged**
